documentation/boltspec_reference.md

```diff
@@ -1225,6 +1186,206 @@ The `allow_upload` function accepts the following modifiers:
   ```ruby
   allow_upload('sshd_config').error_with('msg' => 'Not authorized')
   ```
+  
+## Any stubs
+
+Stubs serve two purposes when you are writing tests for your plans: they allow
+plan functions to be invoked during a plan run and also allow you to set
+their return values. For example, you might use a stub to allow the `run_task`
+function to be invoked any number of times during a test. `allow_any_*` stubs 
+allow you to stub _all_ related function invocations. This can be useful in larger
+plans. For example, `allow_any_task` can be used in place of invoking `run_task` 
+for each individual task.
```

Comment:
```suggestion
Stubs serve two purposes when you are writing tests for your plans: they allow
plan functions to be invoked during a plan run and they allow you to set
return values for functions. For example, you might use a stub to allow the `run_task`
function to be invoked any number of times during a test. `allow_any_*` stubs
allow you to stub all related function invocations. This can be useful in larger
plans. For example, `allow_any_task` can be used in place of invoking `run_task` 
for each individual task.
```